nlbq/core.py

```diff
@@ -60,6 +58,33 @@ def execute(self, query: str) -> Tuple:
         field_names = [field.name for field in results.schema]
         rows = [row.values() for row in results]
         return field_names, rows
+    
+    async def answer(self, question, statement, results) -> str:
+        prompt_messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": self.prompt_template % question},
+            {"role": "assistant", "content": statement},
+            {
+                "role": "system",
+                "content": "The SQL statement returned the following results:\n\n"
+                + results,
+            },
+            {
+                "role": "system",
+                "content": f"Using these results, answer the question: {question}",
+            },
+            {
+                "role": "system",
+                "content": "Don't say what the query was, just answer the question.",
+            },
+        ]
+        resp = await openai.ChatCompletion.acreate(
+            model=self.model,
+            messages=prompt_messages,
+            temperature=0,
+        )
```

Comment:
note: we pass the api_key in #5 where using openapi methods (https://github.com/torchbox/nlbq/pull/5/files#diff-accfc1918290b83c825655513bfc80a420b67cec27062ae406b82b9fa4066267R48)

Reply:
Noted. From memory, that was necessary with async methods and .env loading, right?